great_expectations_cloud/agent/actions/data_assistants/utils.py

```diff
@@ -35,15 +35,15 @@ def build_batch_request(
 ) -> BatchRequest:
     datasource = context.get_datasource(datasource_name=event.datasource_name)
     if not isinstance(datasource, FluentDatasource):
-        raise ValueError(
+        raise TypeError(  # noqa: TRY003 # one off error
```

Comment:
This should be a TypeError
https://docs.astral.sh/ruff/rules/type-check-without-type-error/